api/genericresource/resource_management.go

```diff
@@ -0,0 +1,62 @@
+package genericresource
+
+import (
+	"github.com/docker/swarmkit/api"
+)
+
+// Claim assigns GenericResources to the task by taking them from the node's GenericResource
+// and storing them in the taken store
+func Claim(nodeRes *api.GenericResource, taskRes *api.GenericResource, taken *api.GenericResource) error {
+	t, err := getGenericResourceDiscrete(taskRes)
+	if err != nil {
+		return err
+	}
+
+	switch nr := nodeRes.Resource.(type) {
+	case *api.GenericResource_Discrete:
+		if nr.Discrete < t.Discrete {
+			return NewError("cannot provide enough of scalar resource requested %d, available %d",
+				t.Discrete, nr.Discrete)
+		}
+		nr.Discrete -= t.Discrete
+		taken.Resource = NewDiscrete(t.Discrete)
+	case *api.GenericResource_Set:
+		i := uint64(0)
+		res := NewSet()
+
+		for k := range nr.Set.Val {
+			if i >= t.Discrete {
```

Comment:
If the elements in {GPU-0, GPU-1} have to be equivalent, GPU-0 is as good as GPU-1. Why do users even care which one is assigned?

In docker `cpuset` allows users pin application to a selected CPU. My concern is users might think similar capability is available when they see `set` in generic resources. 

```
dchen@vm2:~/go/src/github.com/docker/docker$ docker run --help | grep -i cpuset-cpus
      --cpuset-cpus string             CPUs in which to allow execution (0-3, 0,1)
```